packages/components/src/popover/index.js

```diff
@@ -515,10 +483,18 @@ const Popover = ( {
 				}
 			) }
 			{ ...contentProps }
-			onKeyDown={ maybeClose }
-			{ ...focusOutsideProps }
-			ref={ mergeRefs( [ containerRef, focusOnMountRef ] ) }
-			tabIndex="-1"
+			ref={ mergeRefs( [ containerRef, dialogRef ] ) }
+			{ ...dialogProps }
+			onKeyDown={ ( event ) => {
+				if ( dialogProps.onKeyDown ) {
+					dialogProps.onKeyDown( event );
```

Comment:
Not sure I understand this. Where's the dialog's onKeyDown handler? Or is this purely to ensure future compatibility?

Reply:
I actually thought we had one in useFocusOutside, now I wonder whether I should keep the future compat or not.

Reply:
Hm, there's also `contentProps`, so for full compatibility we have to merge all handlers in `contentProps` and `dialogProps`. The beauty of not using React but DOM event listeners is that you don't have to merge them, so hopefully there's a way to rewrite `useFocusOutside` to not use React event listeners. :D

Reply:
>  there's a way to rewrite useFocusOutside to not use React event listeners. :D

cc @talldan we're counting on yo :P